### ISSUE TITLE
BRP072C42 Controller controls

### ISSUE THREAD

**Issue by atr00** [rickymarcon/homebridge-daikin#4]
Hi,

On my AC units using BRP072C42, I am not able to change the swing mode and set it to dehumidifier or fan mode.

a get_control_info will give this answer:
`"https://192.168.1.68/aircon/get_control_info"
ret=OK,pow=1,mode=3,adv=,stemp=26.0,shum=0,dt1=26.0,dt2=M,dt3=26.0,dt4=25.0,dt5=25.0,dt7=26.0,dh1=0,dh2=50,dh3=0,dh4=0,dh5=0,dh7=0,dhh=50,b_mode=3,b_stemp=26.0,b_shum=0,alert=255,f_rate=B,b_f_rate=B,dfr1=B,dfr2=A,dfr3=B,dfr4=A,dfr5=A,dfr6=A,dfr7=B,dfrh=A,f_dir=0,b_f_dir=0,dfd1=0,dfd2=0,dfd3=0,dfd4=0,dfd5=0,dfd6=0,dfd7=0,dfdh=0%`

**Comment by rickymarcon** [comment]
Swing mode is controlled by `f_dir` and `b_f_dir`, so if it's not working, can you see if it works by calling the endpoint directly?

`https://192.168.1.68/aircon/set_control_info" ret=OK,pow=1,mode=3,adv=,stemp=26.0,shum=0,dt1=26.0,dt2=M,dt3=26.0,dt4=25.0,dt5=25.0,dt7=26.0,dh1=0,dh2=50,dh3=0,dh4=0,dh5=0,dh7=0,dhh=50,b_mode=3,b_stemp=26.0,b_shum=0,alert=255,f_rate=B,b_f_rate=B,dfr1=B,dfr2=A,dfr3=B,dfr4=A,dfr5=A,dfr6=A,dfr7=B,dfrh=A,f_dir=1,b_f_dir=1,dfd1=0,dfd2=0,dfd3=0,dfd4=0,dfd5=0,dfd6=0,dfd7=0,dfdh=0%`

On the modes, there are only 3 main fan modes you can set through the homebridge plugin; AUTO, HEAT and COOL: https://github.com/homebridge/HAP-NodeJS/blob/a361d8f82ca6e87d90e2aec49bf88d75210fd2e3/src/lib/gen/HomeKit.ts#L2692

There are a few DEHUMIDIFIER controls but I think I need to implement them separately. Haven't investigated whether the Daikin API allows this...

**Comment by atr00** [comment]
Calling the endpoint is working. Swing mode changed accordingly.
Weirdly, I can set the mode back to 0 by turning off oscillate but can't set it back to 1 (turning on oscillate won't change a thing).
I can also set the Aircon to DRY like that so I would think the implementation is fairly easy (as well as set it to FAN mode).

I used [this page](https://github.com/ael-code/daikin-control) for the mapping.

Would be great to have a custom mode that will give us the cooling only AC (auto, dry, cool, fan - remove heat) as well as the control over the 4 fan directions (none, horizontal, vertical, 3d).

**Comment by rickymarcon** [comment]
Yes, that's the API this project is following. However, the only issue is that there are some limitations on what can be supported on HomeKit. So not all controls can be setup. It depends on the actual HomeKit widget. That is, you can code for all the attributes but if the UI in the widget only supports certain controls, then we're stuck.

Now you could probably try bake the settings into the config options but then they would be static and you'd have to restart your device every time you felt like changing a setting, which is pretty inconvenient. 

Let me know what setting you want the most first and I can try add it. Problem is, I'm unable to properly test devices anymore so I'll need to find someway to unit test changes or simply cross my fingers🤞

**Comment by atr00** [comment]
We could maybe add a FAN mode than can be set as either FAN or DRY, I think this plugin does it this way (https://www.npmjs.com/package/homebridge-daikin-local - but does not support https for now) and a fan speed (is that possible? not sure the UI allows that).
As you said it will be static, set in the config file (it's fine to say fan mode will be dehumidifier mode, better than nothing :))

My knowledge of HomeKit is fairly limited but I understand it comes with template limitations: it is not possible to remove the "heat" mode and replace it by a DRY mode right?

If you explain me how to test new additions I will happily help beta test.

**Comment by rickymarcon** [comment]
Sorry for the delay in responding...

Ok, let me take a look at implementing something and you can test :)

**Comment by atr00** [comment]
Excellent :)
